File: frontend/src/app/interfaces/locacao.ts
import { Cliente } from "./cliente"
import { Dependente } from "./dependente"
import { Item } from "./item"
import { Socio } from "./socio"

export interface Locacao{
  id?: number,
  dtLocacao: string,
  dtDevolucaoPrevista: string,
  dtDevolucaoEfetiva: string,
  valorCobrado: number,
  multaCobrada: number,
  cliente: Cliente,
  item: Item
}

File: backend/src/main/java/com/vinicius/locadora/model/Locacao.java
package com.vinicius.locadora.model;

import java.time.LocalDate;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Locacao {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private LocalDate dtLocacao;
    private LocalDate dtDevolucaoPrevista;
    private LocalDate dtDevolucaoEfetiva;
    private Double valorCobrado;
    private Double multaCobrada;

    @JsonIgnore
    @ManyToOne
    private Cliente cliente;
    
    @JsonIgnore
    @ManyToOne
    private Item item;
}

